plugins/clipboard/plugin.js

```diff
@@ -163,10 +190,19 @@
 					// If data empty check for image content inside data transfer. https://dev.ckeditor.com/ticket/16705
 					// Allow both dragging and dropping and pasting images as base64 (#4681).
 					if ( !data && isFileData( evt, dataTransfer ) ) {
-						var file = dataTransfer.getFile( 0 );
+						var file = dataTransfer.getFile( 0 ),
+							defaultSupportedImageTypes = [ 'image/png', 'image/jpeg', 'image/gif' ];
+
+						// Check if the current file type is supported by the default supported types.
+						// We need to check only the image types as other file types require an additional file uploader. (#5095)
+						if ( !isSupportedImageMimeType( file, defaultSupportedImageTypes ) ) {
+							var supportedMimeTypes = CKEDITOR.plugins.clipboard.ignoredImageMimeTypes.concat( defaultSupportedImageTypes );
 
-						if ( CKEDITOR.tools.indexOf( supportedImageTypes, file.type ) === -1 ) {
-							editor.showNotification( unsupportedTypeMsg, 'info', editor.config.clipboard_notificationDuration );
+							if ( isImageType( file ) && !isSupportedImageMimeType( file, supportedMimeTypes ) ) {
```

Comment:
So now, the User sees the notification only for unsupported image files - unless the User tells to ignore them too.

I would add other use cases. The developer will have four options:
- the default one: so display the notification for all extensions that are not supported by default
- display the notification for extensions other than (defaults + added by the user). So the User can tell specifically that he wants to support additionally only certain formats (e.g. `pdf`) <-- This one could be added.
- display the notification for all non-supported images - unless the user excludes them. So the user tell that he will handle all non-image files without explicit enumerating them. Or specific currently non-supporting formats.
- don't display the notification at all - something will handle all non-default extensions <-- This one could be added

That might require some fun with cfg options, but if the user decides to handle all non-default image extensions - he might use some flag for it and don't have to define all possible image extensions. Is that right - or there is something that might break this logic (e.g. in other plugins or sth)?